src/cloudant/document.py

```diff
@@ -351,5 +401,5 @@ def put_attachment(self, attachment, content_type, data, headers=None):
             headers=headers
         )
         resp.raise_for_status()
-
+        self.fetch()
```

Comment:
One other possible option is to have a refresh flag either on the method or the entire object to allow a developer to specifically make the choice about whether they would want to have the local dictionary version auto-refreshed.  We've been kicking that idea around for a little while now at the object level (for both databases and documents) but that be a bit of an effort to implement.